### DISCUSSION TITLE
Configuration for Home Assistant interface through InfluxDB

### DISCUSSION THREAD

**Discussion by benkokes** [jertel/vuegraf#19] [Q&A] [answered]
Hello, and thank you for this interface! I was able to get it happily plotting in Grafana, now I am just trying to understand the correct configuration bits for allowing Home Assistant to read in the data from the InfluxDB.

I am pretty sure I am just not pointing to the right place in InfluxDB, but I just don't know where to look. 
The error: Invalid config for [sensor.influxdb]: required key not provided @ data['queries'][0]['measurement']. Got None. (See /config/configuration.yaml, line 38).

If you have a working configuration.yaml, could you please pasted the appropriate influxdb:  and sensor:   configuration bits as a guide to follow?  It would be appreciated as I am just uncertain how to form the query.

Thank You!

**Comment by jertel** [comment]
Hello, I am not familiar with the Home Assistant influx integration so I don't know if I can be much help. If you can provide more context around what you're doing, some logs, and configuration snippets I might be able to point you in the right direction. Or perhaps the Home Assistant discussion forums might be able to get you going more quickly.

**Comment by benkokes** [comment, marked as answer]
Poking around a little more I was able to get my stuff working. For anyone else setting it up for their first time, here is the relevant part of my Home Assistant configuration.yaml file. I am running all of this on a Synology DS218+, running Docker v18.09.0-0513, Home Assistant core-2021.2.0, HACS plug in.  Edit the configuration.yaml by going into the Docker, select the "containers' option on the left side,  then open the details of the Home Assistant container . At the top tabs, select Terminal, then click the 'Create' button to start a bash session.  within bash CLI, the configuration.yaml should be present and editable. Insert the following code in that file (comments are followed by the '#' in the configuration -- remove the comments. also do not use tabs. Spaces only):

influxdb:
  api_version: 1
  host: IP_address_of_device_running_docker_and_HomeAssistant
  port: 8086  #Home assistant port
  database: name_of_specific_InfluxDB_you_created_to_store_Vue_data
  username: Influxdb_username
  password: Influxdb_password
  ssl: false
  verify_ssl: false
  max_retries: 3
  default_measurement: state


sensor: 
  platform: influxdb
  api_version: 1
  database: name_of_specific_InfluxDB_you_created_to_store_Vue_data
  username: Influxdb_username
  password: Influxdb_password
  queries: 
      - name: VueDatas #this is a friendly name that shows up on the widget
        group_function: last  #Other groupings like 'mean' can be used here
        field: usage
        where: '"device_name" = ''None-1,2,3''' #this should be the default for the Vue
        measurement: '"autogen"."energy_usage"'
